packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -1882,6 +1886,26 @@ export class Bucket extends BucketBase {
     }
   }
 
+  /**
+   * Set the log group on the singleton AutoDeleteObjects provider lambda.
+   * Note: Calling this multiple times will override the previously set log group
+   *
+   * In the case of no buckets with `autoDeleteObjects: true` exist in the stack,
+   * this will throw an exception as no AutoDeleteObjects provider lambda will exist
+   * in the stack.
+   *
+   * @param scope the stack with bucket(s) with `autoDeleteObjects: true`.
+   * @param logGroup the log group to use on the lambda.
+   */
+  public static setAutoDeleteObjectsLogGroup(scope: Construct, logGroup: logs.ILogGroup): void {
+    const provider = AutoDeleteObjectsProvider.getProvider(scope, AUTO_DELETE_OBJECTS_RESOURCE_TYPE);
+
+    if (provider === undefined) {
+      throw new Error('Requires at least one bucket with \'autoDeleteObjects: true\'. None is found.');
+    }
+    provider.configureLambdaLogGroup(logGroup.logGroupName);
```

Comment:
Would this code be problematic eventually? Consider the example where I have the following setups:
```
const bucket = new s3.Bucket(this, 'MyTempFileBucket', {
  removalPolicy: cdk.RemovalPolicy.DESTROY,
  autoDeleteObjects: true,
});

s3.Bucket.setAutoDeleteObjectsLogGroup(this, new logs.LogGroup(this, 'MyLogGroup', {
  logGroupName: 'MyLogGroup',
  retention: logs.RetentionDays.FIVE_YEARS
}))

const bucket2 = new s3.Bucket(this, 'MyTempFileBucket2', {
  removalPolicy: cdk.RemovalPolicy.DESTROY,
  autoDeleteObjects: true,
});
```

The first s3 bucket creation and `setAutoDeleteObjectsLogGroup` will work as expected. However, during the second bucket creation, it will use the existing singleton custom resource function (since it already exists) and a lambda log group is configured already.

Reply:
I have updated the code to use a different implementation. But to answer your question:

The `bucket2` will re-use to the same singleton custom resource function which has been configured to use `MyLogGroup`. This is expected.